src/utils/cache/getWallet.ts

```diff
@@ -38,87 +38,50 @@ export const getWallet = async <TWallet extends EVMWallet>({
     return cachedWallet as TWallet;
   }
 
-  const walletDetails = await getWalletDetails({
-    pgtx,
-    address: walletAddress,
-  });
+  let walletDetails: ParsedWalletDetails;
 
-  if (!walletDetails) {
-    throw createCustomError(
-      `No configured wallet found with address ${walletAddress}`,
-      StatusCodes.BAD_REQUEST,
-      "BAD_REQUEST",
-    );
+  try {
+    walletDetails = await getWalletDetails({
+      pgtx,
+      address: walletAddress,
+    });
+  } catch (e) {
+    if (e instanceof WalletDetailsError) {
+      throw createCustomError(e.message, 400, "BAD_REQUEST");
```

Comment:
Nit: Use StatusCodes everywhere for consistency (it'll be helpful if we're searching for where a certain status code is emitted). Or if you don't like em we should just switch off them entirely and use raw numbers.